pinot-common/pom.xml

```diff
@@ -100,22 +100,31 @@
       <!-- "Plugs in" the Calcite's Parser.jj with the variables present in config.fmpp. These contain the custom rules
            as well as the class to which the custom implementation will get generated -->
       <plugin>
-        <groupId>com.googlecode.fmpp-maven-plugin</groupId>
-        <artifactId>fmpp-maven-plugin</artifactId>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-antrun-plugin</artifactId>
         <executions>
           <execution>
-            <id>generate-fmpp-sources</id>
             <phase>generate-sources</phase>
-            <goals>
-              <goal>generate</goal>
-            </goals>
             <configuration>
-              <cfgFile>${project.build.directory}/codegen/config.fmpp</cfgFile>
-              <outputDirectory>${project.build.directory}/generated-sources</outputDirectory>
-              <templateDirectory>${project.build.directory}/codegen/templates</templateDirectory>
+              <target name="generate-code">
+                <taskdef name="fmpp" classname="fmpp.tools.AntTask" classpathref="maven.compile.classpath"/>
+                <fmpp configuration="${project.build.directory}/codegen/config.fmpp"
+                    sourceRoot="${project.build.directory}/codegen/templates"
+                    outputRoot="${project.build.directory}/generated-sources"/>
+              </target>
             </configuration>
+            <goals>
+              <goal>run</goal>
+            </goals>
           </execution>
         </executions>
+        <dependencies>
+          <dependency>
+            <groupId>net.sourceforge.fmpp</groupId>
+            <artifactId>fmpp</artifactId>
+            <version>${fmpp.version}</version>
```

Comment:
Can we pull this up to the parent pom dependency management?

Reply:
It is only used in this place, but looking at the pinot-project where every version is managed in the root pom, that would make sense. (personally I would also put them in alphabetic order, but that's something for another PR)

Reply:
Rebased PR by fixing the conflicts with latest code. Added the dependency at the root pom as agreed and follow the similar dependency management.

Reply:
@rfscholte  https://github.com/Ekryd/sortpom is the good candidate for managing the clean pom and sorting the dependencies in order. If you have not started will take care of it and raise PR for it.

Reply:
Thanks for reviews @rfscholte @Jackie-Jiang.